### PR TITLE
Product Resource: Check result of `wp_parse_url()` to avoid PHP warnings

### PR DESCRIPTION
**Pull request by n7studios**
## Summary

Prevents PHP warnings reported [here](https://n7studios-workspace.slack.com/archives/C02KFR6N1GF/p1670433002965079), which occur when `wp_parse_url()` (a wrapper for PHP's `parse_url()`)fails to parse a Product's URL.

It's not clear why `wp_parse_url()` (i.e. `parse_url()`) would return false when given a ConvertKit Product URL to parse, as checks in the current code (before this PR) ensure we're not attempting this if no Products exist:
https://github.com/ConvertKit/convertkit-wordpress/blob/017fbdfba05da0ff6b71da732a52b348d10bfeb5/includes/class-convertkit-resource-products.php#L63-L80

However, it's good practice to have better handling and checks for edge cases, rather than blindly assuming it will work every time.

## Testing

- `ResourceProductsTest:testGetCommerceJSURL`: Test that the expected commerce.js URL is returned for a ConvertKit account that has Product(s).
- `ResourceProductsNoDataTest:testGetCommerceJSURL`: Test that no commerce.js URL is returned for a ConvertKit account that has no Product(s).

## Checklist

* [x] I have [written a test](TESTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](TESTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](TESTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](DEVELOPMENT.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](DEVELOPMENT.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

### CODE REVIEW

File: includes/class-convertkit-resource-products.php [hunk 1]
```diff
@@ -74,6 +74,16 @@ public function get_commerce_js_url() {
 		// Parse the URL.
 		$parsed_url = wp_parse_url( $product['url'] );
 
+		// Bail if parsing the URL failed.
+		if ( $parsed_url === false ) {
```
Comment (noelherrick) [inline includes/class-convertkit-resource-products.php:78]: Just wanted to make sure this long form is purposeful (vs. `! $parsed_url`) and explicitly calling out `$parsed_url` might be a string or a bool.

**Review by noelherrick — APPROVED** [review]
